tests/acceptance/features/bootstrap/RoleAssignmentContext.php

```diff
@@ -256,4 +255,19 @@ public function userShouldHaveRole(string $user, string $role): void {
 		$assignmentRoleId = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignments"][0]["roleId"];
 		Assert::assertEquals($this->userGetRoleIdByRoleName($this->featureContext->getAdminUserName(), $role), $assignmentRoleId, "user $user has no role $role");
 	}
+
+	/**
+	 * @Then /^the setting API response should have the role "([^"]*)"$/
+	 *
+	 * @param string $role
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 */
+	public function theSettingApiResponseShouldHaveTheRole(string $role): void {
+		$rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
+		$assignmentRoleId = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignments"][0]["roleId"];
```

Comment:
use this if possible: `$this->featureContext->getJsonDecodedResponse()`
```suggestion
		$assignmentRoleId = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())["assignments"][0]["roleId"];
```